src/Type.hs

```diff
@@ -196,31 +194,24 @@ checkTransition env (U.Transition _ name contract iface@(Interface _ decls) iffs
           negation = U.ENot nowhere $
                         foldl (\acc (U.Case _ e _) -> U.EOr nowhere e acc) (U.BoolLit nowhere False) rest
         in rest `snoc` (if isWild lastCase then U.Case pn negation post else lastCase)
+    -- TODO ensure non-overlapping and exhaustiveness (maybe with elaboration and mandatory wildcard?) 
 
     -- | split case into pass and fail case
```

Comment:
this comment is outdated...

Reply:
I believe we still need to ensure that case branches are non-overlapping and the case analysis exhaustive for things to be sound. 

Non-overlapping is pretty easy to do, if we assume that a branch applies only if the one above does not hold, so I'll implement this for now, and leave an note for exhaustiveness.